### PR TITLE
Feat: Summary department API

### PR DESCRIPTION
**Pull request by larfeq**
- [x] New feature
- [ ] Bug fix
- [ ] High impact

**Work description**:

Created /department endpoints to create, update and list departments used in the Summary Notification application

**Expected result**: Implemented endpoints as defined in RFCS (https://github.com/equinor/fusion-rfcs/blob/feat/summary-api/rfcs/summary-notification-architecture.md)

Link to workitem: [#53765](https://statoil-proview.visualstudio.com/Fusion%20Resource%20Allocation/_workitems/edit/53765)

**Test description**:

- [x] Can be tested
- [x] Automatic tests created / updated
- [x] Local tests are passing

**Checklist**:

- [x] Considered automated tests
- [ ] Considered updating specification / documentation
- [ ] Considered work items
- [ ] Considered security
- [ ] Performed developer testing
- [ ] Checklist finalized / ready for review

### CODE REVIEW

File: src/Fusion.Summary.Api/Controllers/DepartmentsController.cs [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+using Microsoft.AspNetCore.Mvc;
+
+namespace Fusion.Summary.Api.Controllers;
+
+public record PutDepartmentRquest(string DepartmentSapId, string FullDepartmentName, Guid ResourceOwnerAzureUniqueId);
+
+public record GetDepartmentResponse(string departmentSapId, Guid resourceOwnerAzureUniqueId, string fullDepartmentName);
+
+[Authorize]
+[Route("api/[controller]")]
```
Comment (HansDahle) [inline src/Fusion.Summary.Api/Controllers/DepartmentsController.cs:13]: Should align with route patterns established in existing apis and throughout the core apis. 

`[HttpGet("/departments")]` e.g. 

The convention based pattern is too volitile to internal refactoring. When we expose an endpoint we are responsible for maintaining it properly, like versioning and deprecation.

File: src/Fusion.Summary.Api/Controllers/DepartmentsController.cs [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+[ApiController]
+public class DepartmentsController : ControllerBase
+{
+    private readonly IDepartmentService _departmentService;
+
+    public DepartmentsController(IDepartmentService departmentService)
```
Comment (HansDahle) [inline src/Fusion.Summary.Api/Controllers/DepartmentsController.cs:19]: Should try and align design patterns with existing apis perhaps? We have tried to standardise the CQRS pattern. This gives consistency across services for developers and reusability with regards to both concepts and toolings, like audit logging, how to extend with new functionality and debugging..

File: src/Fusion.Summary.Api/Controllers/DepartmentsController.cs [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+    [ProducesResponseType(StatusCodes.Status403Forbidden)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    public async Task<IActionResult> GetDepartmentsV1()
+    {
+
+        #region Authorization
+        // Add authorization
+        #endregion Authorization
```
[the remark this answers is not in the document]
Comment (larfeq) [reply src/Fusion.Summary.Api/Controllers/DepartmentsController.cs:45]: I'm unsure if we need granulated access for this for the time being.

[the remark this answers is not in the document]
Comment (Jonathanio123) [reply src/Fusion.Summary.Api/Controllers/DepartmentsController.cs:45]: If the data that the endpoint is accessible to everyone in Equinor then I guess not needed. Maybe we could just remove the auth section then?

File: src/Fusion.Summary.Api/BaseController.cs [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+    public Task DispatchAsync(IRequest command)
+    {
+        return mediator.Send(command);
+    }
+
+    public Task<TResult> DispatchAsync<TResult>(IRequest<TResult> command)
+    {
+        return mediator.Send(command);
+    }
```
[the remark this answers is not in the document]
Comment (larfeq) [reply src/Fusion.Summary.Api/BaseController.cs:56]: Nope :-) Cleaned it up

File: src/Fusion.Summary.Api/Controllers/DepartmentsController.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (larfeq) [reply src/Fusion.Summary.Api/Controllers/DepartmentsController.cs:45]: What do you think @alelund ? The access is protected and the data exposed is very limited, do we need to further lock it down or can we for the time being keep it as is (no additional access control)?

[the remark this answers is not in the document]
Comment (aleklundeq) [reply src/Fusion.Summary.Api/Controllers/DepartmentsController.cs:45]: I think we should add some authorization here. Even though it maybe that the data exposed is very limited we should make sure that it is secured to only be available for the targeted audience, which in this case would be the function.

If we do it on a later stage, as long as it is done before production

[the remark this answers is not in the document]
Comment (Jonathanio123) [reply src/Fusion.Summary.Api/Controllers/DepartmentsController.cs:45]: Could BeTrustedApplication be good here? We're running under core app reg

[the remark this answers is not in the document]
Comment (aleklundeq) [reply src/Fusion.Summary.Api/Controllers/DepartmentsController.cs:45]: Yes, I think that should be good here.

[the remark this answers is not in the document]
Comment (larfeq) [reply src/Fusion.Summary.Api/Controllers/DepartmentsController.cs]: Resolved!

File: src/Fusion.Summary.Api/Domain/Queries/GetDepartment.cs [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+        public Handler(DatabaseContext context)
+        {
+            _context = context;
+        }
+
+        public async Task<QueryDepartment?> Handle(GetDepartment request, CancellationToken cancellationToken)
+        {
+            // Get all departments 
+            var dbDepartments = await _context.Departments.ToListAsync();
```
Comment (Jonathanio123) [inline src/Fusion.Summary.Api/Domain/Queries/GetDepartment.cs:31]: Could we use FirstOrDefaultAsync here to avoid retrieving all deps

Comment (larfeq) [reply src/Fusion.Summary.Api/Domain/Queries/GetDepartment.cs]: Fixed

**Review by Jonathanio123 — APPROVED** [review]
🚀

**Pull request merged**